barretenberg/cpp/src/barretenberg/bb/main.cpp

```diff
@@ -645,7 +663,11 @@ template <IsUltraFlavor Flavor> void write_vk_honk(const std::string& bytecodePa
     using ProverInstance = ProverInstance_<Flavor>;
     using VerificationKey = Flavor::VerificationKey;
 
-    auto constraint_system = get_constraint_system(bytecodePath);
+    bool honk_recursion = false;
+    if constexpr (IsAnyOf<Flavor, UltraFlavor>) {
+        honk_recursion = true;
```

Comment:
Don't we know that this one will always be honk cause "write_vk_honk"

Reply:
Same for the other honk methods

Reply:
it depends on the flavor. It could be a megahonk flavor as well.